codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/testutil/EventStreamUnmarshallTestCases.kt

```diff
@@ -0,0 +1,274 @@
+            use aws_smithy_eventstream::frame::{Header, HeaderValue, Message, UnmarshallMessage, UnmarshalledMessage};
+            use aws_smithy_types::{Blob, DateTime};
+            use crate::error::*;
+            use crate::model::*;
+
+            fn msg(
+                message_type: &'static str,
+                event_type: &'static str,
+                content_type: &'static str,
+                payload: &'static [u8],
+            ) -> Message {
+                let message = Message::new(payload)
+                    .add_header(Header::new(":message-type", HeaderValue::String(message_type.into())))
+                    .add_header(Header::new(":content-type", HeaderValue::String(content_type.into())));
+                if message_type == "event" {
+                    message.add_header(Header::new(":event-type", HeaderValue::String(event_type.into())))
+                } else {
+                    message.add_header(Header::new(":exception-type", HeaderValue::String(event_type.into())))
+                }
+            }
+            fn expect_event<T: std::fmt::Debug, E: std::fmt::Debug>(unmarshalled: UnmarshalledMessage<T, E>) -> T {
+                match unmarshalled {
+                    UnmarshalledMessage::Event(event) => event,
+                    _ => panic!("expected event, got: {:?}", unmarshalled),
+                }
+            }
+            fn expect_error<T: std::fmt::Debug, E: std::fmt::Debug>(unmarshalled: UnmarshalledMessage<T, E>) -> E {
+                match unmarshalled {
+                    UnmarshalledMessage::Error(error) => error,
+                    _ => panic!("expected error, got: {:?}", unmarshalled),
+                }
+            }
+            """,
+        )
+
+        unitTest(
+            name = "message_with_blob",
+            test = """
+                let message = msg("event", "MessageWithBlob", "application/octet-stream", b"hello, world!");
+                let result = ${format(generator)}().unmarshall(&message);
+                assert!(result.is_ok(), "expected ok, got: {:?}", result);
+                assert_eq!(
+                    TestStream::MessageWithBlob(
+                        MessageWithBlob::builder().data(Blob::new(&b"hello, world!"[..])).build()
+                    ),
+                    expect_event(result.unwrap())
+                );
+            """,
+        )
+
+        if (codegenTarget == CodegenTarget.CLIENT) {
+            unitTest(
+                "unknown_message",
+                """
+                let message = msg("event", "NewUnmodeledMessageType", "application/octet-stream", b"hello, world!");
+                let result = ${format(generator)}().unmarshall(&message);
+                assert!(result.is_ok(), "expected ok, got: {:?}", result);
+                assert_eq!(
+                    TestStream::Unknown,
+                    expect_event(result.unwrap())
+                );
+                """,
+            )
+        }
```

Comment:
Stripping this out of `core` happens in another PR right?

Reply:
There is an issue tracking removing most uses of `CodegenTarget`, but we may not be able to get rid of all of them.